includes/compatibility/civicrm.wpml.php

```diff
@@ -146,207 +288,85 @@ public function basepage_match($is_basepage, $post_id) {
   }
 
   /**
-   * Filters the CiviCRM Base URL for the current language reported by WPML
+   * Filters the CiviCRM Base URL for the current language reported by WPML.
    *
-   * Only filters URLs that point to the front-end/back-end, since WordPress admin URLs are
+   * Only filters URLs that point to the front-end, since WordPress admin URLs are not
    * rewritten by WPML.
    *
-   * @since 5.70
+   * @since 5.72
    *
    * @param str $url The URL as built by CiviCRM.
    * @param bool $admin_request True if building an admin URL, false otherwise.
    * @return str $url The URL as modified by WPML.
    */
   public function base_url_filter($url, $admin_request) {
 
+    global $sitepress, $wpml_url_filters;
+
     // Skip when not defined.
-    if (empty($url)) {
+    if (empty($url) || $admin_request) {
       return $url;
     }
 
-    // Grab WPML language slug.
-    $slug = '';
-    $languages = apply_filters('wpml_active_languages', NULL);
-
-    foreach ($languages as $id => $language) {
-      if ($language['active']) {
-        $slug = $id;
-        break;
-      }
-    }
-
-    if (empty($slug)) {
+    // Determine the language code.
+    $code = apply_filters('wpml_current_language', NULL);
+    if (empty($code)) {
       return $url;
     }
-
-    // Obtain lagnuage negotiation setting
-    $wpml_negotiation = apply_filters('wpml_setting', NULL, 'language_negotiation_type');
-
-    // Build the modified URL.
-    global $sitepress;
-    $raw_url = $this->wpml_remove_language_from_url($url, $sitepress, $wpml_negotiation);
-    $language_url = $sitepress->convert_url($raw_url, $slug);
-
-    return $language_url;
-
-  }
-
-  /**
-   * Setup the rewrite rules for WPML.
-   *
-   * @since 5.70
-   *
-   * @param bool $flush_rewrite_rules True if rules flushed, false otherwise.
-   * @param WP_Post $basepage The Base Page post object.
-   */
-  public function rewrite_rules_wpml($flush_rewrite_rules, $basepage) {
-
-    // Bail if WPML is not present.
```

Comment:
Just wondering why we you removed the quick check to see if WPML is actually installed. I think it might make things more efficient and ensure we don't do unnecessary calls. Unless you are already checking this upstream, but I think I checked at one point and didn't see any check like that.

Reply:
Ahh I see you put it farther up and had responded on the actual issue about my comment. All good! Thanks!